src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold = Math.round(DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        int index = getIndex(key);
+        Node<K, V> processingNode = table[index];
+        if (processingNode == null) {
+            table[index] = new Node<K, V>(key, value, null);
+            size++;
+        } else {
+            while (processingNode.next != null) {
+                if (areKeysEqual(processingNode.key, key)) {
+                    processingNode.value = value;
+                    return;
+                }
+                processingNode = processingNode.next;
+            }
+            if (areKeysEqual(processingNode.key, key)) {
+                processingNode.value = value;
+                return;
```

Comment:
Comment about duplication is not resolved. it is possible to collapse line 26 and line 32 if statements. I suggest to iterate while processingNode != null and make processingNode = processingNode.next; like a loop step. In this case you will be able to iterate over all nodes, not n-1 nodes as in current implementation